modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/DatePicker/DatePicker.es.js

```diff
@@ -217,6 +240,23 @@ export default function DatePicker({
 						yearsCheck={false}
 					/>
 
+					{momentFormatInvalid && (
+						<div
+							className="error-container form-feedback-item mt-1"
+							role="alert"
+						>
+							<ClayAlert
+								className="inline-item inline-item-before"
+								displayType="danger"
+								variant="feedback"
+							>
+								{Liferay.Language.get(
+									'please-enter-a-valid-date'
+								)}
+							</ClayAlert>
+						</div>
+					)}
+
```

Comment:
Hello @vngoliveira and @igor-franca, we already have this implemented within `ReactFieldBase` in a component called `FieldFeedback`, so the only thing we need to do is trigger `FieldFeedback` in the error state, for this we have to define three variables: `displayErrors`, `errorMessage` and `valid`. To do this you can extract these three variables from the `otherProps`, and then create a new state with these three values:

```tsx
const [validField, setValidField] = useState({
    displayErrors,
    errorMessage,
    valid
});
````
Then in the `handleBlur` function you will set the values:

```tsx
	const handleBlur = () => {
		onBlur?.();

		if (!otherProps.required) {
			const isFill = NUMBER_REGEX.test(formattedDate);

			const isValidMomentFormat = moment(
				formattedDate,
				momentFormat,
				true
			).isValid();

			if (!isFill || isValidMomentFormat) {
				setValidField({
					displayErrors,
					errorMessage,
					valid,
				});

				return;
			}

			setValidField({
				displayErrors: true,
				errorMessage: Liferay.Language.get('please-enter-a-valid-date'),
				valid: false,
			});
		}
	};
```

After that, you will need to pass the defined state value to the `ReactFieldBase` Component:

```tsx
...

	return (
		<FieldBase
			displayErrors={validField.displayErrors}
			errorMessage={validField.errorMessage}
			localizedValue={localizedValue}
			name={name}
			readOnly={readOnly}
			type="date"
			valid={validField.valid}
			{...otherProps}
		>
		
....
```
This way we can remove this code and the `momentFormatInvalid` state